research/object_detection/dataset_tools/create_ava_tf_record_for_context.py

```diff
@@ -0,0 +1,400 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Code to download and parse the AVA Actions dataset for TensorFlow models.
+
+The [AVA Actions data set](
+https://research.google.com/ava/index.html)
+is a dataset for human action recognition.
+
+This script downloads the annotations and prepares data from similar annotations
+if local video files are available. The video files can be downloaded
+from the following website:
+https://github.com/cvdfoundation/ava-dataset
+
+Prior to running this script, please run download_and_preprocess_ava.sh to
+download input videos.
+
+Running this code as a module generates the data set on disk. First, the
+required files are downloaded (_download_data) which enables constructing the
+label map. Then (in generate_examples), for each split in the data set, the
+metadata and image frames are generated from the annotations for each sequence
+example (_generate_examples). The data set is written to disk as a set of
+numbered TFRecord files.
+
+Generating the data on disk can take considerable time and disk space.
+(Image compression quality is the primary determiner of disk usage.
+
+If using the Tensorflow Object Detection API, set the input_type field 
+in the input_reader to TF_SEQUENCE_EXAMPLE. 
+
+This data is structured for per-clip action classification where images is
+the sequence of images and labels are a one-hot encoded value. See
+as_dataset() for more details.
+
+Note that the number of videos changes in the data set over time, so it will
+likely be necessary to change the expected number of examples.
+
+The argument video_path_format_string expects a value as such:
+  "/path/to/videos/{0}"
+
+"""
+
+import contextlib
+import csv
+import os
+import random
+import sys
+import zipfile
+import collections
+import glob
+
+from absl import app
+from absl import flags
+from absl import logging
+from six.moves import range
+from six.moves import urllib
+import tensorflow.compat.v1 as tf
+import cv2
+import hashlib
+
+from object_detection.utils import dataset_util
+
+POSSIBLE_TIMESTAMPS = range(902, 1798)
+ANNOTATION_URL = "https://research.google.com/ava/download/ava_v2.2.zip"
+SECONDS_TO_MILLI = 1000
+FILEPATTERN = "ava_actions_%s_1fps_rgb"
+SPLITS = {
+    "train": {
+        "shards": 1000,
+        "examples": 862663,
+        "csv": '',
+        "excluded-csv": ''
+    },
+    "val": {
+        "shards": 100,
+        "examples": 243029,
+        "csv": '',
+        "excluded-csv": ''
+    },
+    #Test doesn't have ground truth, so TF Records can't be created
+    "test": {
+        "shards": 100,
+        "examples": 0,
+        "csv": '',
+        "excluded-csv": ''
+    }
+}
+
+NUM_CLASSES = 80
+
+def feature_list_feature(value):
+  return tf.train.FeatureList(feature=value)
+
```

Comment:
In general this file has a lot of overlap with the file above. Can you consolidate these two and just add an option to produce tf examples with fields needed for context to the above module ? Let me know your thoughts.